portal-bridge/src/beacon_bridge.rs

```diff
@@ -78,17 +75,20 @@ impl BeaconBridge {
     ///  Get and serve the latest beacon data.
     async fn launch_latest(&self) {
         // Current sync committee period known by the bridge
-        let current_period = Arc::new(AtomicUsize::new(0));
+        let mut current_period: usize = 0;
 
         loop {
             let consensus_api = self.api.clone();
             let portal_clients = self.portal_clients.clone();
-            let current_period = Arc::clone(&current_period);
 
-            tokio::spawn(async move {
-                Self::serve_latest(consensus_api, portal_clients, current_period).await;
+            let latest_result = tokio::spawn(async move {
+                Self::serve_latest(consensus_api, portal_clients, current_period).await
             });
 
+            current_period = latest_result
+                .await
```

Comment:
I think this `await` will block the loop until the tokio thread returns a result, which I'm trying to escape.

Reply:
Your concern is that serving this data will last longer than the slot time of 12s? If that's true, I'd argue the bridge is just not working anyway, since it's not keeping up with the chain.